message_processor.go

```diff
@@ -50,34 +81,62 @@ func consumeLog(lf *logPair) {
 			logrus.WithField("id", lf.info.ContainerID).WithError(err).Debug("Ignoring error")
 			dec = protoio.NewUint32DelimitedReader(lf.stream, binary.BigEndian, 1e6)
 		}
-		if sendMessage(lf.splunkl, &buf, lf.info.ContainerID) == false {
+		// Add msg to partial buffer and disable buffer reset flag
+		pBufferSize, err := pBuf.pmsg.Write(buf.Line)
+		pBuf.bufferReset = false
+		if err != nil {
+			logrus.WithField("id", lf.info.ContainerID).WithError(err).WithField("Buffer size:",
+				pBufferSize).Error("Error appending to buffer")
 			continue
 		}
-		if sendMessage(lf.jsonl, &buf, lf.info.ContainerID) == false {
+		// Check for partial buffer timer
```

Comment:
let's try to put the logic in different functions: appendPartial(), handleTimeout()